mesonbuild/interpreter/interpreter.py

```diff
@@ -2242,6 +2242,14 @@ def make_test(self, node: mparser.BaseNode,
         if kwargs['timeout'] <= 0:
             FeatureNew.single_use('test() timeout <= 0', '0.57.0', self.subproject, location=node)
 
+        if kwargs['should_fail'] is not None and kwargs['expected_fail'] is not None:
+            raise InvalidArguments('Tried to use both \'should_fail\' and \'expected_fail\'')
+        elif kwargs['should_fail'] is not None:
+            mlog.deprecation('Use expected_fail instead of should_fail.', location=node)
+            kwargs['expected_fail'] = kwargs['should_fail']
+        elif kwargs['expected_fail'] is None:
+            kwargs['expected_fail'] = False
```

Comment:
I would set a new `expected_fail = kwargs['should_fail']` variable here to avoid mutating the inputs of the current function.

/cc @dcbaker

Reply:
That's probably a good idea, done :)